src/unix/linux-core.c

```diff
@@ -307,8 +310,10 @@ void uv__io_poll(uv_loop_t* loop, int timeout) {
     if (nfds == 0) {
       assert(timeout != -1);
 
-      if (timeout == 0)
+      if (timeout == 0) {
+        uv__trace_end(loop, (uv_trace_info_t*)&trace_info);
```

Comment:
Maybe instead of adding the `uv__trace_end()` call on every return, add a label and use `goto`.